hashing/src/primitives.rs

```diff
@@ -106,6 +111,11 @@ where
     hash_pair(leaf_count_bytes, raw_root)
 }
 
+#[cfg_attr(
+    feature = "std",
+    derive(Debug, schemars::JsonSchema, serde::Serialize, serde::Deserialize,),
```

Comment:
So `IndexedMerkleProof`s can be invalid too.  The validity conditions are:
  -  `index > count` 
  - `indexed_merkle_proof.merkle_proof.len() ≠ indexed_merkle_proof.compute_expected_proof_length()`

Reply:
Added in 700422489824098027324d3495eedc12e5845fbc